### ISSUE TITLE
PR to keras

### ISSUE THREAD

**Issue by keunwoochoi** [bckenstler/CLR#1]
Hi, this callback seems quite interesting. Do you plan to PR to the keras repo?

**Comment by bckenstler** [comment]
You know, I'm actually a novice at open source contributions. I'm not sure
exactly how to do that, or whether my callback is interesting enough to
merit a contribution. I would love to hear your thoughts!

**Comment by keunwoochoi** [comment]
I also hadn't had any experience until a couple of months ago, and actually it was really good experience PR-ing some stuffs. There are unfamiliar aspects that you should know/follow about doing a PR, which I think is quite similar to how people work in the industry these days, and which I was lack of. I'd recommend it to everyone unless s/he is extremely busy.

I think it is interesting enough, got no idea if it's interesting enough for others though. We can always discuss on keras issue channel/twitter. I'll ask @fchollet, too.

**Comment by neiljohari** [comment]
Hey! I found this repo after reading Smith's paper, and imo this is definitely worthy of a PR. I'm currently in high school so I lack the ability to say whether or not CLR is widely applicable, but from what the paper is stating this seems like a practical tool for any Neural Network project. It's definitely useful to my research project, and I'm sure many others will appreciate this code a lot more if it's in the core of keras. Worst case, your PR is rejected, and that's not a huge deal. 

Here's how you can open a PR while meeting all of their guidelines:

- The first thing to look at is to see if there's any special instructions for contributing. A glance at the README.md points us to [this contribution guideline file](https://github.com/fchollet/keras/blob/master/CONTRIBUTING.md)
   - There is a section on `Pull Requests`, which tells us how to identify which branch to contribute to and what guidelines must be met
- Go ahead and fork fchollet/keras to your own account
- Clone your own fork
- Add your CLR code to the appropriate place ([this seems like a good spot](https://github.com/fchollet/keras/blob/master/keras/callbacks.py))
- Go ahead and make sure you meet all the bullet points in that CONTRIBUTING.md file
   - At a glance, it seems like you're going to be contributing on the `keras-contrib` branch, and you're going to want to write tests for your new feature by adding to [this file](https://github.com/fchollet/keras/blob/master/tests/keras/test_callbacks.py)


I really hope this all works out. Let me know if you need any help working on the fork/PR, I'd love to help out!

**Comment by gewoonrik** [comment]
I agree with everyone above. This seems like a nice addition to Keras!

**Comment by rivershadowy** [comment]
I agree. This is excellent work that should be appreciated and let more people know about.

BTW, it improved my model a lot. :)

**Comment by maxfrei750** [comment]
@bckenstler Is there any chance, that you will tackle the PR into Keras? If you are not willing to do it yourself, would you allow others to do so?

**Comment by YumainOB** [comment]
I'm sure your implementation is worthy enough as it is referenced in this [pyimagesearch blog post](https://www.pyimagesearch.com/2019/07/29/cyclical-learning-rates-with-keras-and-deep-learning/). And as far as I'm concerned it's a nice reference!

EDIT: My bad I just saw that you are in contributors on this [keras-contrib callback](https://github.com/keras-team/keras-contrib/blob/master/keras_contrib/callbacks/cyclical_learning_rate.py). Is this callback as reliable as this repos implementation? if yes this issue may be closed to stat this isn't it?
